powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java

```diff
@@ -31,89 +34,62 @@
 import io.getlime.security.powerauth.app.tppengine.model.response.GiveConsentResponse;
 import io.getlime.security.powerauth.app.tppengine.model.response.TppAppDetailResponse;
 import io.getlime.security.powerauth.app.tppengine.model.response.UserConsentDetailResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
-import org.springframework.http.converter.HttpMessageConverter;
-import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
-import org.springframework.web.client.HttpStatusCodeException;
-import org.springframework.web.client.ResourceAccessException;
-import org.springframework.web.client.RestTemplate;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
 
-import java.io.IOException;
-import java.util.*;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * TPP Engine Client provides methods for communication with the TPP registry and consent engine.
- * It uses the RestTemplate class to handle REST API calls. Apache HTTP client is used instead of default client
- * so that error responses contain full response bodies.
+ * It uses the Rest Client to handle REST API calls.
  *
  * @author Petr Dvorak, petr@wultra.com
+ * @author Roman Strobl, roman.strobl@wultra.com
  */
 public class TppEngineClient {
 
-    private final String serviceUrl;
-    private final ObjectMapper objectMapper;
-    private final RestTemplate restTemplate;
+    private static final Logger logger = LoggerFactory.getLogger(TppEngineClient.class);
 
-    /**
-     * Create a new client with provided base URL.
-     *
-     * @param serviceUrl Base URL.
-     */
-    public TppEngineClient(String serviceUrl) {
-        this(serviceUrl, null);
-    }
+    private final RestClient restClient;
 
     /**
-     * Create a new client with provided base URL and custom object mapper.
+     * Create a new client with provided base URL.
      *
-     * @param serviceUrl Base URL.
-     * @param objectMapper Object mapper.
+     * @param serviceUrl Base service URL.
+     * @throws TppEngineClientException Thrown when REST client initialization fails.
      */
-    public TppEngineClient(String serviceUrl, ObjectMapper objectMapper) {
-        this.serviceUrl = serviceUrl;
-        if (objectMapper != null) {
-            this.objectMapper = objectMapper;
-        } else {
-            this.objectMapper = objectMapper();
+    public TppEngineClient(String serviceUrl) throws TppEngineClientException {
```

Comment:
I think we need to keep an option to pass `ObjectMapper` here - maybe just introduce a wrapper method?

Reply:
We can use the constructor `TppEngineClient(RestClientConfiguration config)`, which allows setting the custom `ObjectMapper` as a parameter. I think this solves are customization use cases.

Reply:
👍  Yes, this works for me.